### PR TITLE
resource/aws_rds_cluster: Add validateRdsIdentifier for source_cluster_identifier in restore_to_point_in_time block

### PR DESCRIPTION
**Pull request by shuheiktgw**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
* resource/aws_rds_cluster: Add validateRdsIdentifier for source_cluster_identifier in restore_to_point_in_time block
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSRDSCluster_PointInTimeRestore'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSRDSCluster_PointInTimeRestore -timeout 120m
=== RUN   TestAccAWSRDSCluster_PointInTimeRestore
=== PAUSE TestAccAWSRDSCluster_PointInTimeRestore
=== CONT  TestAccAWSRDSCluster_PointInTimeRestore
--- PASS: TestAccAWSRDSCluster_PointInTimeRestore (386.72s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	388.621s
```

You cannot use `validateRdsIdentifier` for validating `source_cluster_identifier` in `restore_to_point_in_time` block. That is because as the document blow suggests, `source_cluster_identifier` can be a full ARN of the source cluster when cross-account cloning an Aurora cluster.

> Clone the cluster by specifying the full ARN of the source cluster in the SourceDBClusterIdentifier parameter of the RDS API operation RestoreDBClusterToPointInTime. If the ARN passed as the SourceDBClusterIdentifier hasn't been shared, then the same error is returned as if the specified cluster doesn't exist.

https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.Managing.Clone.html

In order to handle the case, I've defined a new validation function called `validateRestoreToPointInTimeSourceClusterIdentifier`. Thank you for your review!

### CODE REVIEW

File: aws/resource_aws_rds_cluster.go [hunk 1]
```diff
@@ -232,7 +232,7 @@ func resourceAwsRDSCluster() *schema.Resource {
 							Type:         schema.TypeString,
 							Required:     true,
 							ForceNew:     true,
-							ValidateFunc: validateRdsIdentifier,
+							ValidateFunc: validateRestoreToPointInTimeSourceClusterIdentifier,
```
Comment (anGie44) [inline aws/resource_aws_rds_cluster.go:235]: The validations in the new func look like what we need! I would just suggest using `validation.Any()` here to incorporate the 2 conditions (like an `OR`) e.g.
```suggestion
							ValidateFunc: validation.Any(
							    validateArn,
							    validateRdsIdentifier),
```

**Review by anGie44 — CHANGES_REQUESTED** [review]
Hi @shuheiktgw , thank you for providing this fix! Just a quick suggestion regarding the validation usage in the schema. It would also be great if we could  get in an acceptance test for cross-account cloning with this change 👍 if you have any questions please let me know or are crunched for time to add the test lmk

File: aws/resource_aws_rds_cluster.go (hunk 1 above)
Comment (shuheiktgw) [reply aws/resource_aws_rds_cluster.go:235]: Thank you, I didn't know we already have such a useful function 👍

**Comment by anGie44** [comment]
@shuheiktgw , did the docs provide good example? since this is a straightforward fix, happy to merge this in to get it released and then follow-up with an acctest PR if an issue is raised 👍

**Review by anGie44 — APPROVED** [review]

**Comment by shuheiktgw** [comment]
@anGie44 Sounds great! Let me work on an acctest in a different PR 👍

**Pull request merged**